core/src/main/java/org/apache/iceberg/CatalogType.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+public enum CatalogType {
+  HIVE("org.apache.iceberg.hive.HiveCatalog"),
+  HADOOP("org.apache.iceberg.hadoop.HadoopCatalog"),
+  GLUE("org.apache.iceberg.aws.glue.GlueCatalog"),
+  NESSIE("org.apache.iceberg.nessie.NessieCatalog"),
+  DYNAMODB("org.apache.iceberg.aws.dynamodb.DynamoDbCatalog"),
+  JDBC("org.apache.iceberg.jdbc.JdbcCatalog");
+
+  private final String catalogImpl;
+
+  CatalogType(String catalogImpl) {
+    this.catalogImpl = catalogImpl;
+  }
+
+  public String getTypeName() {
+    return name().toLowerCase(Locale.ENGLISH);
+  }
+
+  public String getCatalogImpl() {
+    return catalogImpl;
```

Comment:
Given that `catalog-impl` is still a configuration value, is the naming of this method (and the corresponding private field name) possibly a bit confusing?

I think it would be possible to avoid having this entirely, as well as the private field, and use a static method like `CatalogType.from(String inputType)`. This would convey the same thing but give a bit more separation from the existing config value of `catalog-impl`?

Is there somewhere that I missed that we need to pass around a stateful / instantiated `CatalogType` outside of a single method call? Possible I just need more coffee 😅